modules/pipeline/aop/plugins/task/plugins/autotest_cookie_keep_before/plugin.go

```diff
@@ -44,45 +47,58 @@ func (p *Plugin) Handle(ctx *aoptypes.TuneContext) error {
 	// will only fetch the latest one
 	reportSets, err := ctx.SDK.Report.GetPipelineReportSet(ctx.SDK.Pipeline.ID, autotest_cookie_keep_after.ReportTypeAutotestSetCookie)
 	if err != nil {
+		rlog.TErrorf(ctx.SDK.Pipeline.ID, ctx.SDK.Task.ID, "failed to get pipeline reports, err: %", err)
 		return err
 	}
-	var cookieJar string
+	var setCookieJSON string
 	for _, v := range reportSets.Reports {
-		if v.Meta == nil || v.Meta[autotest_cookie_keep_after.CookieJar] == nil {
+		if v.Meta == nil || v.Meta[apitestsv2.HeaderSetCookie] == nil {
 			continue
 		}
-		cookieJar = v.Meta[autotest_cookie_keep_after.CookieJar].(string)
+		setCookieJSON = v.Meta[apitestsv2.HeaderSetCookie].(string)
 		break
 	}
-
-	if len(cookieJar) <= 0 {
+	rlog.TDebugf(ctx.SDK.Pipeline.ID, ctx.SDK.Task.ID, "setCookieJSON: %s", setCookieJSON)
+	if setCookieJSON == "" {
+		return nil
+	}
+	// parse Set-Cookie-JSON to Cookie
+	var setCookies []string
+	if err := json.Unmarshal([]byte(setCookieJSON), &setCookies); err != nil {
+		return fmt.Errorf("failed to parse Set-Cookie: %s, err: %v", setCookieJSON, err)
+	}
+	if len(setCookies) == 0 {
 		return nil
 	}
+	setCookie := setCookies[0]
 
-	logrus.Infof("autotest keep cookie： %v", cookieJar)
+	logrus.Infof("pipelineID: %d, taskID: %d, autotest keep cookie: %v", ctx.SDK.Pipeline.ID, ctx.SDK.Task.ID, setCookie)
 	// if autoTestAPIConfig is empty
 	// means not use config to run, also need to keep cookie
 	var config apistructs.AutoTestAPIConfig
-	if len(ctx.SDK.Task.Extra.PrivateEnvs[autotest_cookie_keep_after.AutotestApiGlobalConfig]) >= 0 {
-		err := json.Unmarshal([]byte(ctx.SDK.Task.Extra.PrivateEnvs[autotest_cookie_keep_after.AutotestApiGlobalConfig]), &config)
-		if err != nil {
+	if configStr, ok := ctx.SDK.Task.Extra.PrivateEnvs[autotest_cookie_keep_after.AutotestApiGlobalConfig]; ok {
+		if err := json.Unmarshal([]byte(configStr), &config); err != nil {
+			rlog.TErrorf(ctx.SDK.Pipeline.ID, ctx.SDK.Task.ID, "failed to unmarshal AUTOTEST_API_GLOBAL_CONFIG, err: %v", err)
 			return err
 		}
 	}
 	if config.Header == nil {
 		config.Header = map[string]string{}
 	}
-	config.Header[autotest_cookie_keep_after.CookieJar] = cookieJar
-	configJson, err := json.Marshal(config)
+	config.Header[apitestsv2.HeaderCookie] = setCookie
```

Comment:
if config user set cookie in header, here you will cover user cookie

Reply:
Overwrite cookie can implement user switch.